fine_tuning_embeddings/fine_tune_embed_model.py

```diff
@@ -0,0 +1,499 @@
+    generate_qa_embedding_pairs,
+    EmbeddingQAFinetuneDataset,
+    SentenceTransformersFinetuneEngine,
+)
+from llama_index.embeddings import OpenAIEmbedding
+from sentence_transformers.evaluation import InformationRetrievalEvaluator
+from sentence_transformers import SentenceTransformer
+from llama_index import SimpleDirectoryReader, ServiceContext, VectorStoreIndex
+from llama_index.node_parser import SentenceSplitter
+from llama_index.schema import MetadataMode, TextNode
+from llama_index.llms import LangChainLLM, OpenAI
+from typing import List, Union
+import pickle  # Import for saving and loading .pkl files
+import os
+from pathlib import Path
+import dotenv
+import sys
+
+sys.path.append("..")
+import src.models.sambanova_endpoint
+
+
+# Setup logging
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+# Process Environment Variables
+dotenv.load_dotenv()
+
+
+def split_files_into_datasets(
+    input_data_directory: str,
+    output_data_directory: str,
+    file_extension: str = "pdf",
+    split_ratio: float = 0.8,
+) -> Tuple[str, str]:
+    """
+    Split files in the given directory into training and validation datasets based on the split ratio and save them to the specified output directory.
+
+    Args:
+    - input_data_directory: The directory path to search for files.
+    - file_extension: The type of files to search for, defaults to 'pdf'.
+    - split_ratio: The ratio to split the files for training, with the rest for validation.
+    - output_data_directory: The directory where processed files are stored.
+
+    Returns:
+    - Tuple containing paths to the generated CSV files for training and validation datasets.
+    """
+
+    files = glob.glob(f"{input_data_directory}/**/*.{file_extension}", recursive=True)
+    shuffle(files)
+
+    print(files)
+
+    split_index = int(len(files) * split_ratio)
+    train_files = files[:split_index]
+    val_files = files[split_index:]
+
+    train_csv_path = os.path.join(output_data_directory, "train_files.csv")
+    val_csv_path = os.path.join(output_data_directory, "val_files.csv")
+
+    # Make output dir
+    Path(output_data_directory).mkdir(parents=True, exist_ok=True)
+
+    for dataset, path in [(train_files, train_csv_path), (val_files, val_csv_path)]:
+        with open(path, "w", newline="", encoding="utf-8") as csvfile:
+            writer = csv.writer(csvfile)
+            writer.writerow(["path", "filename"])
+            for file in dataset:
+                writer.writerow([file, os.path.basename(file)])
+
+    logging.info(f"Datasets generated: Train - {train_csv_path}, Val - {val_csv_path}")
+    return train_csv_path, val_csv_path
+
+
+# Function to instantiate LLM (Example with OpenAI, can be modified for others)
+def instantiate_llm():
+    # Initialize LLM to be used in generating queries for fine tuning.
+    # Example LLM instantiation:
+    # For a Sambanova LLM:
+    base_url = "YOUR_BASE_URL"
```

Comment:
Please use export.env for this.
Some examples in the other starter-kits.

from dotenv import load_dotenv
load_dotenv('../../export.env')

Reply:
export.env now being used.